src/kernels/kernel.ts

```diff
@@ -191,7 +191,6 @@ abstract class BaseKernel implements IBaseKernel {
             Array.from(this.hooks.get('willInterrupt') || new Set<Hook>()).map((h) => h())
         );
         traceInfo(`Interrupt requested ${getDisplayPath(this.resourceUri || this.uri)}`);
-        this.startCancellation.cancel();
```

Comment:
This alone will not work, as the previous code line 280 was used to cancel the pending start, hence cancelling the previous and immediately initializing it with a new item will not have the desired effect of the cancellation.

Personally not sure why we're cancellign either, but this is existing code and reluctant to change that at this stage, will look into that in next iteration debt.